components/supervisor/cmd/init.go

```diff
@@ -114,18 +114,22 @@ var initCmd = &cobra.Command{
 			}
 		}()
 		// start the reaper to clean up zombie processes
+		reaperChan := make(chan reaper.Status, 10)
 		reaper.Start(reaper.Config{
 			Pid:              -1,
 			Options:          0,
 			DisablePid1Check: false,
-			OnReap: func(pid int, wstatus syscall.WaitStatus) {
-				if pid != runCommand.Process.Pid {
-					return
+			StatusChannel:    reaperChan,
+		})
+		go func() {
+			for status := range reaperChan {
```

Comment:
```suggestion
			for status, more := range reaperChan {
			         if !more {
			                 return
			         }
```

Reply:
There's no seconds return for `range <chan>` case https://go.dev/play/p/SgY7x8Citwf because range will finish itself

(But `status, more := <- reaperChan` case has seconds return)

Reply:
> because range will finish itself

Ah, thank you! Makes sense :pray: